docs/developers/code-style-guide.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Karmada Code Style Guide
+---
+
+Karmada is primarily developed in Go and follows the standard Go community coding conventions. Automated style checks are 
+enforced via [golangci-lint](https://github.com/karmada-io/karmada/blob/e2c4b596a5da442fc0dbeab9f9063d8db8669208/.github/workflows/ci.yml#L18-L35).
+However, the resulting code can still look and feel very differently among different developers. To ensure consistency 
+across the codebase, this guide defines additional rules beyond linter defaults.
+
+Adhering to this guide helps new contributors onboard quickly, reduces PR review iterations, shortens maintainer review 
+cycles, and ultimately speeds up merge velocity.
+
+> Note: This guide is not a replacement for the official Go style guide but rather a supplement to it. All aspects not 
+> covered in this guide should follow the official Go style guide and best practices.
+> Additionally, existing code in the Karmada codebase may not fully comply with this guide. We encourage adherence to
+> this guide when writing new code or modifying existing code to gradually improve code quality and consistency.
+
+## Code Documentation and Commenting
+
+### All exported functions, methods, structs, and interfaces **must** be documented with clear and concise comments describing their purpose and behavior.
```

Comment:
Titles should not be too long; as you can see in the preview, it doesn't look right.
We may summarize the title and then elaborate on the details in the main text.

Same as below.